### PR TITLE
Create Background service

### PR DESCRIPTION
**Pull request by Finka95**
Service works in the background and closes open tickets every 5 minutes, where last message from User who has UserRole.Operator 

And moved close ticket logic from repo to ticketService

### CODE REVIEW

**Review by hinkevich — APPROVED** [review]

File: TicketSystem/TicketSystem/BackgroundServices/TicketTimedHostedService.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            _serviceProvider = services;
+            _logger = logger;
+        }
+
+        protected override Task ExecuteAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Scoped service running");
+            _timer = new Timer(DoWork, null, TimeSpan.Zero, TimeSpan.FromMinutes(5));
+            return Task.CompletedTask;
```
Comment (5Ka-me) [inline TicketSystem/TicketSystem/BackgroundServices/TicketTimedHostedService.cs]: Add empty lines

File: TicketSystem/TicketSystem/BackgroundServices/TicketTimedHostedService.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                var ticketService = scope.ServiceProvider.GetRequiredService<ITicketService>();
+                ticketService.CloseOpenTickets();
+            }
+        }
+
+        public override async Task StopAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Scoped service is stopping.");
+            _timer?.Change(Timeout.Infinite, 0);
+            await base.StopAsync(cancellationToken);
```
Comment (5Ka-me) [inline TicketSystem/TicketSystem/BackgroundServices/TicketTimedHostedService.cs]: Add empty lines

File: TicketSystem/TicketSystem/Data/Repositories/TicketRepository.cs [hunk 3]
```diff
@@ -52,27 +52,17 @@ public async Task<IEnumerable<Ticket>> GetAllAsync(CancellationToken cancellatio
...
 
-            foreach (var ticket in tickets)
-            {
-                if (ticket.CreatedAt.AddMinutes(minutesToClose) < DateTime.Now)
-                    ticket.TicketStatus = TicketStatus.Closed;
-            }
-
-            await _context.SaveChangesAsync(cancellationToken);
-        }
+        //}
```
Comment (5Ka-me) [inline TicketSystem/TicketSystem/Data/Repositories/TicketRepository.cs]: Delete comment

**Review by Genresal — APPROVED** [review]

File: TicketSystem/TicketSystem/Program.cs [hunk 4]
```diff
@@ -1,10 +1,12 @@
+using TicketSystem.BackgroundServices;
 using TicketSystem.Extensions;
 using TicketSystem.Services.DI;
 
 var builder = WebApplication.CreateBuilder(args);
 var configuration = new ConfigurationBuilder().AddJsonFile("appsettings.json").Build();
 
 builder.Services.AddBusinessLogicLayerServices(configuration);
+builder.Services.AddHostedService<TicketTimedHostedService>();
```
Comment (5Ka-me) [inline TicketSystem/TicketSystem/Program.cs]: You can use method chaining

File: TicketSystem/TicketSystem/Services/TicketService.cs [hunk 5]
```diff
@@ -58,5 +43,22 @@ public async Task<IEnumerable<Ticket>> GetTicketsAsync(CancellationToken cancell
...
+            var tickets = await _ticketRepository.GetTicketsByConditionsAsync(cancellationToken,
+                t => t.TicketStatus == TicketStatus.Open
+                     && t.Messages.Any()
+                     && t.Messages.Last().User.UserRole == UserRole.Operator);
+
+            foreach (var ticket in tickets)
+            {
+                if (ticket.CreatedAt.AddMinutes(MinutesToClose) < DateTime.Now)
+                    ticket.TicketStatus = TicketStatus.Closed;
+
```
Comment (5Ka-me) [inline TicketSystem/TicketSystem/Services/TicketService.cs]: Remove empty line

**Review by CrazyKamenwik — APPROVED** [review]

**Review by 5Ka-me — APPROVED** [review]

**Pull request merged**